docs/concepts/tx-lifecycle.md

```diff
@@ -0,0 +1,130 @@
+			  v				   v				   v
+		-----------------------		-----------------------		-----------------------
+States synced:  | CheckTxState(t+1)   |         | DeliverTxState(t+1) |		|   QueryState(t+1)   |
+		-----------------------		|                     |		|		      |
+		          |			|                     |		|		      |
+			  v			|                     |		|	  	      |
+			  .				   .				   .
+			  .				   . 				   .
+			  .				   .  				   .
+			  
+```
+### PostChecks
+`CheckTx` also returns `GasUsed` which may or may not be less than the user's provided `GasWanted`. A `PostCheckFunc` is an optional filter run after `CheckTx` that can be used to enforce that users provide enough gas. 
+
+
+## Consensus
+At each round, a proposer is chosen amongst the validators to create and propose the next block. This validator (presumably honest) has generated a Mempool of validated transactions and now includes them in a block. The validators execute [Tendermint BFT Consensus](https://tendermint.com/docs/spec/consensus/consensus.html#terms); with 2/3 approval from the validators, the block is committed. Note that not all blocks have the same number of transactions and it is possible for consensus to result in a nil block or one with no transactions - here, it is assumed that the transaction has made it this far. 
+
+## State Changes
+During consensus, the validators came to agreement on not only which transactions but also the precise order of the transactions. However, apart from committing to this block in consensus, the ultimate goal is actually for nodes to commit to a new state generated by the transaction state changes. 
+In order to execute the transaction, the following ABCI function calls are made in order to communicate to the application what state changes need to be made. While nodes each run everything individually, since the messages' state transitions are deterministic and the order was finalized during consensus, this process yields a single, unambiguous result.
+```
+		-----------------------		
+		| BeginBlock	      |        
+		-----------------------		
+		          |		
+			  v			
+		-----------------------		    
+		| DeliverTx(tx0)      |  
+		| DeliverTx(tx1)      |   	  
+		| DeliverTx(tx2)      |  
+		| DeliverTx(tx3)      |  
+		|	.	      |  
+		|	.	      |
+		|	.	      |
+		-----------------------		
+		          |			
+			  v			
+		-----------------------
+		| EndBlock	      |         
+		-----------------------
+		          |			
+			  v			
+		-----------------------
+		| Commit	      |         
+		-----------------------
+```
+#### BeginBlock
+`BeginBlock` is run first, and mainly transmits important data such as block header and Byzantine Validators from the last round of consensus to be used during the next few steps. No transactions are handled here.
+
+#### DeliverTx
+The `DeliverTx` ABCI function defined in [`baseapp`](https://github.com/cosmos/cosmos-sdk/blob/8b1d75caa2099800ee9983e4a4528bcd00fec302/baseapp/baseapp.go
```

Comment:
do not link to `baseapp` codebase here, but to `baseapp` concept doc. Only the concept doc of component X should directly reference code of component X